minwoo.lee_java/src/BOJ1277.java

```diff
@@ -0,0 +1,84 @@
+import java.io.*;
+import java.util.*;
+
+public class BOJ1277 {
+    static int N, W;
+    static int[][] location;
+    static double M;
+    static boolean[][] connect;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        N = Integer.parseInt(st.nextToken());
+        W = Integer.parseInt(st.nextToken());
+        M = Double.parseDouble(br.readLine());
+
+        // 각 발전소의 좌표 값 저장
+        location = new int[N + 1][];
+        for (int i = 1; i <= N; i++) {
+            st = new StringTokenizer(br.readLine());
+            int x = Integer.parseInt(st.nextToken());
+            int y = Integer.parseInt(st.nextToken());
+            location[i] = new int[]{x, y};
+        }
+
+        // 전선이 있고 있는 것을 표시
+        connect = new boolean[N + 1][N + 1];
+        for (int i = 0; i < W; i++) {
+            st = new StringTokenizer(br.readLine());
+            int from = Integer.parseInt(st.nextToken());
+            int to = Integer.parseInt(st.nextToken());
+            connect[from][to] = true;
+            connect[to][from] = true;
+        }
+        dijkstra();
+    }
+
+
+    private static void dijkstra() {
+        PriorityQueue<Node> pq = new PriorityQueue<>((o1, o2) -> {
+            return (int) (o1.dist * 1000) - (int) (o2.dist * 1000);
+        });
```

Comment:
이 람다는 아래와 같이 변경할 수 있습니다. 그리고 우선순위 큐에서 비교할 때 1000을 곱하고 정수로 만들어서 비교하나 distance 그대로 비교하나 어짜피 결과는 동일하지 않을까요?
```suggestion
        PriorityQueue<Node> pq = new PriorityQueue<>(Comparator.comparingInt(o -> (int) (o.dist * 1000)));
```

Reply:
생각해보니 말씀해주신대로 1000을 곱하지 않고 비교를 해도 되는군요.
그리고 변경해주신 코드를 보면서 `interface`가 가지고 있는 `static` 메소드들에 대해서도 공부해서 활용할 수 있도록 하겠습니다.
api 문서를 확인해보니 `Comparator.comparingDouble` 사용하면 더 정확한 비교를 할 수 있을 것 같습니다.